src/providers/tpm_provider/key_management.rs

```diff
@@ -114,11 +115,24 @@ impl TpmProvider {
         app_name: ApplicationName,
         op: psa_import_key::Operation,
     ) -> Result<psa_import_key::Result> {
-        if op.attributes.key_type != Type::RsaPublicKey {
-            error!("The TPM provider currently only supports importing RSA public key.");
-            return Err(ResponseStatus::PsaErrorNotSupported);
+        match op.attributes.key_type {
+            Type::RsaPublicKey => self.psa_import_key_internal_rsa_public(app_name, op),
+            Type::RsaKeyPair => self.psa_import_key_internal_rsa_keypair(app_name, op),
+            _ => {
+                error!(
+                    "The TPM provider does not support the {:?} key type.",
```

Comment:
As far as I know, the TPM provider does not support *importing* these other key types but it might support generating keys of those types